logpolicy/logpolicy.go

```diff
@@ -554,9 +576,9 @@ func NewWithConfigPath(collection, dir, cmdName string, netMon *netmon.Monitor,
 		PrivateID:    newc.PrivateID,
 		Stderr:       logWriter{console},
 		CompressLogs: true,
-		HTTPC:        &http.Client{Transport: NewLogtailTransport(logtail.DefaultHost, netMon, logf)},
+		HTTPC:        cmp.Or(opts.HTTPC, &http.Client{Transport: NewLogtailTransport(logtail.DefaultHost, opts.NetMon, logf)}),
```

Comment:
instead of initializing this here and then resetting it below, move this to an else block on 595?